src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -29,9 +29,10 @@ public class WebHostBuilder : IWebHostBuilder
         private readonly List<Action<ILoggerFactory>> _configureLoggingDelegates;
 
         private IConfiguration _config;
-        private ILoggerFactory _loggerFactory;
         private WebHostOptions _options;
         private bool _webHostBuilt;
+        private Func<IHostingEnvironment, IConfiguration, ILoggerFactory> _loggerFactoryFunc;
+        private Action<IConfigurationBuilder, IHostingEnvironment> _configureConfigurationBuilderDelegate;
```

Comment:
I didn't do it because ordering of configuration providers is so important. Allowing multiples means that extension methods can add a config provider to the very end of the list of providers, overriding keys from any added before it. I am not against it completely, but it wasn't immediately obvious to me that it was a good thing.